lib/src/map/map_controller.dart

```diff
@@ -117,179 +127,12 @@ abstract class MapController {
-  /// based on the map's current screen positioning
-  CustomPoint<double> latLngToScreenPoint(LatLng mapCoordinate);
-
-  CustomPoint<double> rotatePoint(
-    CustomPoint mapCenter,
-    CustomPoint point, {
-    bool counterRotation = true,
-  });
-
-  /// Current center coordinates
-  LatLng get center;
-
-  /// Current outer points/boundaries coordinates
-  LatLngBounds? get bounds;
-
-  /// Current zoom level
-  double get zoom;
-
-  /// Current rotation in degrees, where 0° is North
-  double get rotation;
+  /// Current FlutterMapState.
+  FlutterMapState get mapState;
 
   /// [Stream] of all emitted [MapEvent]s
   Stream<MapEvent> get mapEventStream;
 
-  /// Underlying [StreamSink] of [mapEventStream]
-  ///
-  /// Usually prefer to use [mapEventStream].
-  StreamSink<MapEvent> get mapEventSink;
-
-  /// Immediately change the internal map state
-  ///
-  /// Not recommended for external usage.
-  set state(FlutterMapState state);
-
-  /// Dispose of this controller by closing the [mapEventStream]'s
-  /// [StreamController]
-  ///
-  /// Not recommended for external usage.
+  /// Dispose of this controller.
   void dispose();
 }
-
-@internal
-class MapControllerImpl implements MapController {
-  MapControllerImpl._();
-
-  @override
-  bool move(
-    LatLng center,
-    double zoom, {
-    Offset offset = Offset.zero,
-    String? id,
-  }) =>
-      _state.move(
-        center,
-        zoom,
-        offset: offset,
-        id: id,
-        source: MapEventSource.mapController,
-      );
-
-  @override
-  bool rotate(double degree, {String? id}) =>
-      _state.rotate(degree, id: id, source: MapEventSource.mapController);
-
-  @override
-  MoveAndRotateResult rotateAroundPoint(
-    double degree, {
-    CustomPoint<double>? point,
-    Offset? offset,
-    String? id,
-  }) =>
-      _state.rotateAroundPoint(
-        degree,
-        point: point,
-        offset: offset,
-        id: id,
-        source: MapEventSource.mapController,
-      );
-
-  @override
-  MoveAndRotateResult moveAndRotate(
-    LatLng center,
-    double zoom,
-    double degree, {
-    String? id,
-  }) =>
-      _state.moveAndRotate(
-        center,
-        zoom,
-        degree,
-        source: MapEventSource.mapController,
-        id: id,
-      );
-
-  @override
-  bool fitBounds(
-    LatLngBounds bounds, {
-    FitBoundsOptions? options =
-        const FitBoundsOptions(padding: EdgeInsets.all(12)),
-  }) =>
-      _state.fitBounds(bounds, options!);
-
-  @override
-  CenterZoom centerZoomFitBounds(
-    LatLngBounds bounds, {
-    FitBoundsOptions? options =
-        const FitBoundsOptions(padding: EdgeInsets.all(12)),
-  }) =>
-      _state.centerZoomFitBounds(bounds, options!);
-
-  @override
-  LatLng pointToLatLng(CustomPoint localPoint) =>
-      _state.pointToLatLng(localPoint);
-
-  @override
-  CustomPoint<double> latLngToScreenPoint(LatLng latLng) =>
-      _state.latLngToScreenPoint(latLng);
```

Comment:
These need to be restored.

Reply:
It's now available via the `mapController.mapState.centerZoomFitBounds` etc. If we want to avoid breaking changes I can add those methods back with deprecations?

Reply:
I'm not sure why they need to be moved in the first place? I guess because it doesn't actually change any state?

Reply:
I have restored them as deprecated methods, what do you think? In my opinion accessing the state via a getter in a controller is an anti-pattern as it makes the control flow bi-directional (MapController <-> Map state/frame). It would be preferable to either:

-  If in a child of FlutterMap use FlutterMapState.of(context) (now MapFrame.of(context)) which will ensure the widget gets rebuilt when the frame changes.
- If not in a child widget, use a callback (we would have to add onFrameChanged) which the user can use with a Stream that they manage themselves. i.e. (onFrameChanged: (oldFrame, newFrame) => _frameStream.add(newFrame)).

That way the control flow is unidirectional which is easier to reason about. (MapController -> FlutterMap -> MapFrame -> children/listeners). At the same time there are examples of Flutter widgets that have bi-directional flow with their controller e.g. TextEditingController 🤷.  Anway I don't want to make this change (it's a huge breaking change for users) but I did want to reduce the number of methods which expose the state from the controller so that the MapFrame itself becomes the main API for accessing the state of the map.

If the general opinion is that it's better to leave the old methods without deprecations I'm not too worried about changing it back.